src/related.tsx

```diff
@@ -175,6 +175,7 @@ class Related extends KalturaPlayer.core.BasePlugin {
    * @memberof Related
    */
   addRelatedListComponents() {
+    let isEnded = false;
```

Comment:
i suggest adding a new method for adding the events handlers and for setting isEnded, and calling the method from loadMedia, to separate event handler logic from and the logic for injecting components into the ui

also, because you need to clear event listeners anyway on player reset (it's best practice against memory leaks, but also because loadMedia will be called again on every entry change, like in playlist, and it will add more event listeners every time it's called), i suggest using this.eventManager.listen to add the listeners, and then reset() when will call this.eventManager.removeAll it will remove them for you